ETH-Merge-Bots/Changes-in-Block-Difficulty/src/agent.ts

```diff
@@ -0,0 +1,74 @@
+import { Finding, HandleBlock, BlockEvent, getEthersProvider, ethers } from "forta-agent";
+import { NUMBER_OF_BLOCKS_TO_CHECK, FINDING_THRESHOLD, createFinding, TERMINAL_TOTAL_DIFFICULTY } from "./utils";
+import BigNumber from "bignumber.js";
+
+BigNumber.set({ DECIMAL_PLACES: 2 });
+
+export const provideHandleBlock = (
+  init: boolean,
+  provider: ethers.providers.Provider,
+  numberOfBlocksToCheck: number,
+  threshold: BigNumber
+): HandleBlock => {
+  let blockDifficulties: BigNumber[] = new Array<BigNumber>();
+  let blockDifficultyMovingAverage: BigNumber;
+  let blockDifficultiesSum: BigNumber = new BigNumber(0);
+
+  return async (blockEvent: BlockEvent) => {
+    const findings: Finding[] = [];
+
+    // check if TTD has been reached
+    if (blockEvent.block.totalDifficulty >= TERMINAL_TOTAL_DIFFICULTY) return findings;
```

Comment:
This condition will never be true because `totalDifficulty` is hex strings and `TERMINAL_TOTAL_DIFFICULTY` is number.

Reply:
Oops